context/context.go

```diff
@@ -207,7 +207,11 @@ func (c *fsContext) Branch() (string, error) {
 	}
 
 	currentBranch, err := git.CurrentBranch()
-	if err != nil {
+	switch err {
+	case nil:
+	case git.ErrNotOnAnyBranch:
```

Comment:
I don't think that you need to make any changes to `ctx.Branch()` implementation at all. Instead, you can allow the original error get wrapped like before. Then, at places where you check for a specific error, you can use `errors.Is()`.